temporal/fx.go

```diff
@@ -345,6 +447,7 @@ func FrontendServiceProvider(
 }
 
 func WorkerServiceProvider(
+	cfg *config.Config,
```

Comment:
we need to define a initParams struct that leverage fx.In and put all these input as fields in there.

Reply:
will do in followup PR